lightning/src/chain/channelmonitor.rs

```diff
@@ -1418,21 +1417,24 @@ impl<Signer: EcdsaChannelSigner> ChannelMonitor<Signer> {
 		};
 
 		let onchain_tx_handler = OnchainTxHandler::new(
-			channel_value_satoshis, channel_keys_id, destination_script.into(), keys,
-			channel_parameters.clone(), initial_holder_commitment_tx, secp_ctx
+			channel_parameters.channel_value_satoshis, channel_keys_id, destination_script.into(),
+			keys, channel_parameters.clone(), initial_holder_commitment_tx, secp_ctx
```

Comment:
Will `OnchainTxHandler` hold parameters for each funding scope?

Unrelated to this PR, but I wonder if we need to write `channel_parameters` twice if we can pass it to `OnchainTxHandler` on read.